### ISSUE TITLE
Error starting daemon: Error initializing network controller: list bridge addresses failed: no available network

### ISSUE THREAD

**Issue by borysn** [docker/docker#31546]
hey guys,

i am building docker from source on gentoo. pulling from app-emulation/docker. building runs fine. no kernel config or use flag issues. running `contrib/check-config.sh` checks out OK, just missing AUFS_FS and some zfs stuff which is fine.

when trying to start the systemd docker service, i receive the following error...

```
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Sat 2017-03-04 17:08:30 CST; 54min ago
     Docs: https://docs.docker.com
  Process: 1964 ExecStart=/usr/bin/dockerd -H fd:// (code=exited, status=1/FAILURE)
 Main PID: 1964 (code=exited, status=1/FAILURE)
      CPU: 142ms

Mar 04 17:08:28 abnd dockerd[1964]: time="2017-03-04T17:08:28.402272665-06:00" level=info msg="libcontainerd: new containerd process, pid: 1988"
Mar 04 17:08:29 abnd dockerd[1964]: time="2017-03-04T17:08:29.553155958-06:00" level=info msg="[graphdriver] using prior storage driver: overlay2"
Mar 04 17:08:29 abnd dockerd[1964]: time="2017-03-04T17:08:29.797022584-06:00" level=info msg="Graph migration to content-addressability took 0.00 seconds"
Mar 04 17:08:29 abnd dockerd[1964]: time="2017-03-04T17:08:29.813577762-06:00" level=info msg="Loading containers: start."
Mar 04 17:08:29 abnd dockerd[1964]: time="2017-03-04T17:08:29.871799411-06:00" level=info msg="Firewalld running: false"
Mar 04 17:08:30 abnd dockerd[1964]: Error starting daemon: Error initializing network controller: list bridge addresses failed: no available network
Mar 04 17:08:30 abnd systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
Mar 04 17:08:30 abnd systemd[1]: Failed to start Docker Application Container Engine.
Mar 04 17:08:30 abnd systemd[1]: docker.service: Unit entered failed state.
Mar 04 17:08:30 abnd systemd[1]: docker.service: Failed with result 'exit-code'
```

i am currently running 

docker-1.13.1-r2
docker-runc-1.0.0_rc2_20170201
docker-proxy-0.8.0_p20161111
containerd-0.2.5-r1
systemd-233

i am not sure how to resolve the list bridge addresses failed issue. it seems like this could just be an environment issue, do i need to create my own network bridge for docker? is the ebuild install process on gentoo missing something?

i'm just at a loss here. any help would be great.

**Comment by aboch** [comment]
Is your host connected to VPN ?
If so, because of the gateway redirect route, docker won't be able to elect one of its default address pools to assign to the docker0 interface.

**Comment by borysn** [comment]
@aboch yea i saw that, i am not using a vpn though

**Comment by borysn** [comment]
I'm wondering if there is an undocumented use flag or kernel option necessary for use with systemd. or possibly i overlooked a bug already reported else where. Could it be a missing dependency, or maybe some arbitrary environment issue? This one's got me scratching my head.

**Comment by aboch** [comment]
If you are blocked by this, a workaround is to manually create the `docker0` bridge interface, then start the docker daemon. Or start the docker daemon passing an explicit bridge ip/subnet (ex. `dockerd --bip 192.168.100.1/24`)

PTAL at #30295 although you should not be affected since you are already on 1.13.1

**Comment by borysn** [comment]
@aboch yea i was confused, i thought upgrading would have resolved these issues for me. thanks for the replies. i'll give the proposed solutions a go. probably just write up a script to create the docker0 bridge using bridge-utils.

**Comment by borysn** [comment]
@aboch alright, so creating the docker0 resolved my issue. Apparently systemd is having some bugs with cgroups, so i also had to add `systemd.legacy_systemd_cgroup_controller=yes` to my kernel boot options.

here is the bash script i used to create the docker0, actually just slightly modified from the [build your own bridge](https://docs.docker.com/engine/userguide/networking/default_network/build-bridges/) user guide.

``` bash
#!/bin/bash

#
# create docker0 bridge
# restart docker systemd service
# confirm new outgoing NAT masquerade is set up
#
# reference
#     https://docs.docker.com/engine/userguide/networking/default_network/build-bridges/
#


sudo brctl addbr docker0
sudo ip addr add 192.168.42.1/24 dev docker0
sudo ip link set dev docker0 up
ip addr show docker0
sudo systemctl restart docker
sudo iptables -t nat -L -n

exit(0)
```

i'm going to go ahead and close this one up.

**Comment by yinjun622** [comment]
@borysn hi~
after you added bridge
can docker image be run?

**Comment by yinjun622** [comment]
I can not run any images.
log:

`
[root@bidb003-db-jp2v-dev:/etc]# dockerd
INFO[0000] libcontainerd: new containerd process, pid: 4073
INFO[0001] [graphdriver] using prior storage driver: overlay
INFO[0001] Graph migration to content-addressability took 0.00 seconds
INFO[0001] Loading containers: start.
INFO[0001] Firewalld running: false
INFO[0001] Default bridge (docker0) is assigned with an IP address 192.168.5.0/24. Daemon option --bip can be used to set a preferred IP address
INFO[0001] Loading containers: done.
INFO[0001] Daemon has completed initialization
INFO[0001] Docker daemon                                 commit=3a232c8 graphdriver=overlay version=17.03.0-ce
INFO[0001] API listen on /var/run/docker.sock
^CINFO[0017] Processing signal 'interrupt'
INFO[0017] stopping containerd after receiving terminated
[root@bidb003-db-jp2v-dev:/etc]# dockerd -h
Flag shorthand -h has been deprecated, please use --help

Usage:  dockerd COMMAND

A self-sufficient runtime for containers.

Options:
      --add-runtime runtime                   Register an additional OCI compatible runtime (default [])
      --api-cors-header string                Set CORS headers in the Engine API
      --authorization-plugin list             Authorization plugins to load (default [])
      --bip string                            Specify network bridge IP
  -b, --bridge string                         Attach containers to a network bridge
      --cgroup-parent string                  Set parent cgroup for all containers
      --cluster-advertise string              Address or interface name to advertise
      --cluster-store string                  URL of the distributed storage backend
      --cluster-store-opt map                 Set cluster store options (default map[])
      --config-file string                    Daemon configuration file (default "/etc/docker/daemon.json")
      --containerd string                     Path to containerd socket
      --cpu-rt-period int                     Limit the CPU real-time period in microseconds
      --cpu-rt-runtime int                    Limit the CPU real-time runtime in microseconds
  -D, --debug                                 Enable debug mode
      --default-gateway ip                    Container default gateway IPv4 address
      --default-gateway-v6 ip                 Container default gateway IPv6 address
      --default-runtime string                Default OCI runtime for containers (default "runc")
      --default-ulimit ulimit                 Default ulimits for containers (default [])
      --disable-legacy-registry               Disable contacting legacy registries
      --dns list                              DNS server to use (default [])
      --dns-opt list                          DNS options to use (default [])
      --dns-search list                       DNS search domains to use (default [])
      --exec-opt list                         Runtime execution options (default [])
      --exec-root string                      Root directory for execution state files (default "/var/run/docker")
      --experimental                          Enable experimental features
      --fixed-cidr string                     IPv4 subnet for fixed IPs
      --fixed-cidr-v6 string                  IPv6 subnet for fixed IPs
  -g, --graph string                          Root of the Docker runtime (default "/var/lib/docker")
  -G, --group string                          Group for the unix socket (default "docker")
      --help                                  Print usage
  -H, --host list                             Daemon socket(s) to connect to (default [])
      --icc                                   Enable inter-container communication (default true)
      --init                                  Run an init in the container to forward signals and reap processes
      --init-path string                      Path to the docker-init binary
      --insecure-registry list                Enable insecure registry communication (default [])
      --ip ip                                 Default IP when binding container ports (default 0.0.0.0)
      --ip-forward                            Enable net.ipv4.ip_forward (default true)
      --ip-masq                               Enable IP masquerading (default true)
      --iptables                              Enable addition of iptables rules (default true)
      --ipv6                                  Enable IPv6 networking
      --label list                            Set key=value labels to the daemon (default [])
      --live-restore                          Enable live restore of docker when containers are still running
      --log-driver string                     Default driver for container logs (default "json-file")
  -l, --log-level string                      Set the logging level ("debug", "info", "warn", "error", "fatal") (default "info")
      --log-opt map                           Default log driver options for containers (default map[])
      --max-concurrent-downloads int          Set the max concurrent downloads for each pull (default 3)
      --max-concurrent-uploads int            Set the max concurrent uploads for each push (default 5)
      --metrics-addr string                   Set default address and port to serve the metrics api on
      --mtu int                               Set the containers network MTU
      --oom-score-adjust int                  Set the oom_score_adj for the daemon (default -500)
  -p, --pidfile string                        Path to use for daemon PID file (default "/var/run/docker.pid")
      --raw-logs                              Full timestamps without ANSI coloring
      --registry-mirror list                  Preferred Docker registry mirror (default [])
      --seccomp-profile string                Path to seccomp profile
      --selinux-enabled                       Enable selinux support
      --shutdown-timeout int                  Set the default shutdown timeout (default 15)
  -s, --storage-driver string                 Storage driver to use
      --storage-opt list                      Storage driver options (default [])
      --swarm-default-advertise-addr string   Set default address or interface for swarm advertised address
      --tls                                   Use TLS; implied by --tlsverify
      --tlscacert string                      Trust certs signed only by this CA (default "/root/.docker/ca.pem")
      --tlscert string                        Path to TLS certificate file (default "/root/.docker/cert.pem")
      --tlskey string                         Path to TLS key file (default "/root/.docker/key.pem")
      --tlsverify                             Use TLS and verify the remote
      --userland-proxy                        Use userland proxy for loopback traffic (default true)
      --userland-proxy-path string            Path to the userland proxy binary
      --userns-remap string                   User/Group setting for user namespaces
  -v, --version                               Print version information and quit

Run 'dockerd COMMAND --help' for more information on a command.
[root@bidb003-db-jp2v-dev:/etc]# dockerd -l debug
DEBU[0000] docker group found. gid: 112
DEBU[0000] Listener created for HTTP on unix (/var/run/docker.sock)
INFO[0000] libcontainerd: new containerd process, pid: 4228
DEBU[0000] libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable
DEBU[0001] Using default logging driver json-file
DEBU[0001] Golang's threads limit set to 57330
INFO[0001] [graphdriver] using prior storage driver: overlay
DEBU[0001] Using graph driver overlay
DEBU[0001] Max Concurrent Downloads: 3
DEBU[0001] Max Concurrent Uploads: 5
INFO[0001] Graph migration to content-addressability took 0.00 seconds
INFO[0001] Loading containers: start.
DEBU[0001] Option Experimental: false
DEBU[0001] Option DefaultDriver: bridge
DEBU[0001] Option DefaultNetwork: bridge
INFO[0001] Firewalld running: false
DEBU[0001] /usr/sbin/iptables, [--wait --version]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D OUTPUT -m addrtype --dst-type LOCAL ! --dst 127.0.0.0/8 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D OUTPUT -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D PREROUTING]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D OUTPUT]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -F DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -X DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -F DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -X DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -F DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -X DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -n -L DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -N DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -n -L DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -n -L DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C DOCKER-ISOLATION -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -I DOCKER-ISOLATION -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C POSTROUTING -s 192.168.5.0/24 ! -o docker0 -j MASQUERADE]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C DOCKER -i docker0 -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -I DOCKER -i docker0 -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -i docker0 -o docker0 -j DROP]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -j DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -I FORWARD -j DOCKER-ISOLATION]
DEBU[0001] Network (b6f5fd2) restored
DEBU[0001] Allocating IPv4 pools for network bridge (b6f5fd2ffe0c218e2f0768e73d04b84d8e6967c1e1a63e44d5f7b9930271b1c2)
DEBU[0001] RequestPool(LocalDefault, 192.168.5.0/24, , map[], false)
DEBU[0001] RequestAddress(LocalDefault/192.168.5.0/24, 192.168.5.1, map[RequestAddressType:com.docker.network.gateway])
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C POSTROUTING -s 192.168.5.0/24 ! -o docker0 -j MASQUERADE]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D POSTROUTING -s 192.168.5.0/24 ! -o docker0 -j MASQUERADE]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C DOCKER -i docker0 -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -D DOCKER -i docker0 -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -o docker0 -j DOCKER]
DEBU[0001] releasing IPv4 pools from network bridge (b6f5fd2ffe0c218e2f0768e73d04b84d8e6967c1e1a63e44d5f7b9930271b1c2)
DEBU[0001] ReleaseAddress(LocalDefault/192.168.5.0/24, 192.168.5.1)
DEBU[0001] ReleasePool(LocalDefault/192.168.5.0/24)
INFO[0001] Default bridge (docker0) is assigned with an IP address 192.168.5.0/24. Daemon option --bip can be used to set a preferred IP address
DEBU[0001] Allocating IPv4 pools for network bridge (4b753e0a81c7f6eb329125b78b3f799baf27c8e6fe3f29b19f12c77279564349)
DEBU[0001] RequestPool(LocalDefault, 192.168.5.0/24, , map[], false)
DEBU[0001] RequestAddress(LocalDefault/192.168.5.0/24, 192.168.5.1, map[RequestAddressType:com.docker.network.gateway])
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C POSTROUTING -s 192.168.5.0/24 ! -o docker0 -j MASQUERADE]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -I POSTROUTING -s 192.168.5.0/24 ! -o docker0 -j MASQUERADE]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C DOCKER -i docker0 -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -I DOCKER -i docker0 -j RETURN]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -i docker0 -o docker0 -j DROP]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -I FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -I FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -I FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0001] /usr/sbin/iptables, [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -I FORWARD -o docker0 -j DOCKER]
DEBU[0001] /usr/sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -D FORWARD -j DOCKER-ISOLATION]
DEBU[0001] /usr/sbin/iptables, [--wait -I FORWARD -j DOCKER-ISOLATION]
INFO[0001] Loading containers: done.
INFO[0001] Daemon has completed initialization
INFO[0001] Docker daemon                                 commit=3a232c8 graphdriver=overlay version=17.03.0-ce
DEBU[0001] Registering routers
DEBU[0001] Registering GET, /containers/{name:.*}/checkpoints
DEBU[0001] Registering POST, /containers/{name:.*}/checkpoints
DEBU[0001] Registering DELETE, /containers/{name}/checkpoints/{checkpoint}
DEBU[0001] Registering HEAD, /containers/{name:.*}/archive
DEBU[0001] Registering GET, /containers/json
DEBU[0001] Registering GET, /containers/{name:.*}/export
DEBU[0001] Registering GET, /containers/{name:.*}/changes
DEBU[0001] Registering GET, /containers/{name:.*}/json
DEBU[0001] Registering GET, /containers/{name:.*}/top
DEBU[0001] Registering GET, /containers/{name:.*}/logs
DEBU[0001] Registering GET, /containers/{name:.*}/stats
DEBU[0001] Registering GET, /containers/{name:.*}/attach/ws
DEBU[0001] Registering GET, /exec/{id:.*}/json
DEBU[0001] Registering GET, /containers/{name:.*}/archive
DEBU[0001] Registering POST, /containers/create
DEBU[0001] Registering POST, /containers/{name:.*}/kill
DEBU[0001] Registering POST, /containers/{name:.*}/pause
DEBU[0001] Registering POST, /containers/{name:.*}/unpause
DEBU[0001] Registering POST, /containers/{name:.*}/restart
DEBU[0001] Registering POST, /containers/{name:.*}/start
DEBU[0001] Registering POST, /containers/{name:.*}/stop
DEBU[0001] Registering POST, /containers/{name:.*}/wait
DEBU[0001] Registering POST, /containers/{name:.*}/resize
DEBU[0001] Registering POST, /containers/{name:.*}/attach
DEBU[0001] Registering POST, /containers/{name:.*}/copy
DEBU[0001] Registering POST, /containers/{name:.*}/exec
DEBU[0001] Registering POST, /exec/{name:.*}/start
DEBU[0001] Registering POST, /exec/{name:.*}/resize
DEBU[0001] Registering POST, /containers/{name:.*}/rename
DEBU[0001] Registering POST, /containers/{name:.*}/update
DEBU[0001] Registering POST, /containers/prune
DEBU[0001] Registering PUT, /containers/{name:.*}/archive
DEBU[0001] Registering DELETE, /containers/{name:.*}
DEBU[0001] Registering GET, /images/json
DEBU[0001] Registering GET, /images/search
DEBU[0001] Registering GET, /images/get
DEBU[0001] Registering GET, /images/{name:.*}/get
DEBU[0001] Registering GET, /images/{name:.*}/history
DEBU[0001] Registering GET, /images/{name:.*}/json
DEBU[0001] Registering POST, /commit
DEBU[0001] Registering POST, /images/load
DEBU[0001] Registering POST, /images/create
DEBU[0001] Registering POST, /images/{name:.*}/push
DEBU[0001] Registering POST, /images/{name:.*}/tag
DEBU[0001] Registering POST, /images/prune
DEBU[0001] Registering DELETE, /images/{name:.*}
DEBU[0001] Registering OPTIONS, /{anyroute:.*}
DEBU[0001] Registering GET, /_ping
DEBU[0001] Registering GET, /events
DEBU[0001] Registering GET, /info
DEBU[0001] Registering GET, /version
DEBU[0001] Registering GET, /system/df
DEBU[0001] Registering POST, /auth
DEBU[0001] Registering GET, /volumes
DEBU[0001] Registering GET, /volumes/{name:.*}
DEBU[0001] Registering POST, /volumes/create
DEBU[0001] Registering POST, /volumes/prune
DEBU[0001] Registering DELETE, /volumes/{name:.*}
DEBU[0001] Registering POST, /build
DEBU[0001] Registering POST, /swarm/init
DEBU[0001] Registering POST, /swarm/join
DEBU[0001] Registering POST, /swarm/leave
DEBU[0001] Registering GET, /swarm
DEBU[0001] Registering GET, /swarm/unlockkey
DEBU[0001] Registering POST, /swarm/update
DEBU[0001] Registering POST, /swarm/unlock
DEBU[0001] Registering GET, /services
DEBU[0001] Registering GET, /services/{id}
DEBU[0001] Registering POST, /services/create
DEBU[0001] Registering POST, /services/{id}/update
DEBU[0001] Registering DELETE, /services/{id}
DEBU[0001] Registering GET, /services/{id}/logs
DEBU[0001] Registering GET, /nodes
DEBU[0001] Registering GET, /nodes/{id}
DEBU[0001] Registering DELETE, /nodes/{id}
DEBU[0001] Registering POST, /nodes/{id}/update
DEBU[0001] Registering GET, /tasks
DEBU[0001] Registering GET, /tasks/{id}
DEBU[0001] Registering GET, /secrets
DEBU[0001] Registering POST, /secrets/create
DEBU[0001] Registering DELETE, /secrets/{id}
DEBU[0001] Registering GET, /secrets/{id}
DEBU[0001] Registering POST, /secrets/{id}/update
DEBU[0001] Registering GET, /plugins
DEBU[0001] Registering GET, /plugins/{name:.*}/json
DEBU[0001] Registering GET, /plugins/privileges
DEBU[0001] Registering DELETE, /plugins/{name:.*}
DEBU[0001] Registering POST, /plugins/{name:.*}/enable
DEBU[0001] Registering POST, /plugins/{name:.*}/disable
DEBU[0001] Registering POST, /plugins/pull
DEBU[0001] Registering POST, /plugins/{name:.*}/push
DEBU[0001] Registering POST, /plugins/{name:.*}/upgrade
DEBU[0001] Registering POST, /plugins/{name:.*}/set
DEBU[0001] Registering POST, /plugins/create
DEBU[0001] Registering GET, /networks
DEBU[0001] Registering GET, /networks/
DEBU[0001] Registering GET, /networks/{id:.+}
DEBU[0001] Registering POST, /networks/create
DEBU[0001] Registering POST, /networks/{id:.*}/connect
DEBU[0001] Registering POST, /networks/{id:.*}/disconnect
DEBU[0001] Registering POST, /networks/prune
DEBU[0001] Registering DELETE, /networks/{id:.*}
INFO[0001] API listen on /var/run/docker.sock

`

**Comment by thaJeztah** [comment]
@yinjun622 please open a new issue with the details requested in the issue reporting template; the original issue here was resolved.

**Comment by linuxadmin25** [comment]
On WSL2,

After running sudo dockerd , I ended upon ::

failed to start daemon: Error initializing network controller: Error creating default "bridge" network: permission denied

Any thoughts how to fix this issue?

Regards,
Tanmay
